python/pyspark/sql/pandas/serializers.py

```diff
@@ -226,7 +229,12 @@ def _create_array(self, series, arrow_type, spark_type=None):
         else:
             mask = series.isnull()
         try:
-            return pa.Array.from_pandas(series, mask=mask, type=arrow_type, safe=self._safecheck)
+            if arrow_cast:
+                return pa.Array.from_pandas(series, mask=mask, type=arrow_type).cast(
+                    target_type=arrow_type, safe=self._safecheck
+                )
+            else:
+                return pa.Array.from_pandas(series, mask=mask, safe=self._safecheck)
```

Comment:
@xinrong-meng `type=arrow_type` is missing here.

Reply:
#41795

Reply:
Thanks!